tests/src/Kernel/Entity/EntityGroupsTest.php

```diff
@@ -0,0 +1,216 @@
+
+    $actual = Og::getEntityGroups('user', $this->user2->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user2->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user1->id()));
+  }
+
+  /**
+   * Tests other groups users are added to.
+   */
+  public function testOtherGroups() {
+    // Should be a part of no groups.
+    $this->assertEquals([], Og::getEntityGroups('user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+
+    Og::invalidateCache();
+
+    // Add to group 1 should return that.
+    $this->createMembership($this->user3, $this->group1);
+
+    $actual = Og::getEntityGroups('user', $this->user3->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+
+    Og::invalidateCache();
+
+    // Add to group 2 should also return that.
+    $this->createMembership($this->user3, $this->group2);
+
+    $actual = Og::getEntityGroups('user', $this->user3->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(2, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertTrue(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+  }
+
+  /**
+   * Creates an Og membership entity,
+   *
+   * @param \Drupal\user\Entity\User $user
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *
+   * @return \Drupal\og\Entity|OgMembership
+   */
+  protected function createMembership($user, $group) {
+    $membership = OgMembership::create(['type' => OG_MEMBERSHIP_TYPE_DEFAULT])
+      ->setEntityId($user->id())
+      ->setEntityType('user')
+      ->setGid($group->id())
+      ->setGroupType($group->getEntityTypeId())
+      ->save();
+
+    return $membership;
+  }
+
+  /**
+   * Asserts whether a group ID exists in some results.
+   *
+   * Assumes entity_type is used.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group_to_check
+   * @param array $results
+   */
+  protected function assertGroupExistsInResults($group_to_check, array $results) {
```

Comment:
Oh, do you mean in the closure? If we have a $group param, we can use $group_entity - sure!